eos-reclaim-swap

```diff
@@ -0,0 +1,122 @@
+        if (p[-1] == '/dev/mapper/endless-image'):
+            disk = partition[:-1]
+            self.sep = ''
+
+        # for testing with loop devices
+        if (p[:-3] == '/dev/loop'):
+            self.sep = 'p'
+
+        # for testing with image files
+        if not p.startswith('/dev/'):
+            disk = partition[:-1]
+            self.sep = ''
+
+        return disk
+
+
+    def reclaim_swap_space(self, disk_path):
+        '''
+        Remove the first swap partition on disk and expand the preceding
+        partition so it spawns over the space previously used by the swap
+        partition.
+        '''
+        dev = parted.getDevice(disk_path)
+        disk = parted.newDisk(dev)
+        parts = disk.partitions
+
+        for p in range(len(parts)):
+            # (py)parted does not expose the MBR partition ID or the GPT GUID,
+            # so we need to use the file system signature to identify swap
+            # partitions and thus this only works with formatted swap
+            # partitions. When the fileSystem property is set on swap
+            # partitions it can be either linux-swap(v0) or linux-swap(v1).
+            if parts[p].fileSystem and parts[p].fileSystem.type[:-4] == 'linux-swap':
+                geom = parted.Geometry(start=parts[p-1].geometry.start,
+                                       end=parts[p].geometry.end, device=dev)
+                constraint = parted.Constraint(maxGeom=geom)
+                disk.removePartition(parts[p])
+                disk.maximizePartition(parts[p-1], constraint)
+                disk.commit()
+
+                return True
+
+        return False
+
+
+    def find_disk_and_reclaim_swap_space(self, part, unit):
+        if not part:
+            part = subprocess.run(['systemctl', 'show', '-p', 'What', unit],
+                                  stdout=subprocess.PIPE).stdout
+            part = part.decode()
+            part = part.split(sep='=')[1].strip()
+
+        disk = self.find_disk(part)
+        if (disk == None):
+            raise ValueError('Could not find disk for partition %s' % part)
+        if self.verbose: print('Old disk table:\n' + self.get_table_str(disk))
+
+        removed = self.reclaim_swap_space(disk)
+        if (removed and os.geteuid() == 0):
+            # schedule filesystem resize for next boot
+            os.remove('/var/eos-booted')
+        if self.verbose: print('New disk table:\n' + self.get_table_str(disk))
+
+
+    def __init__(self, verbose=False):
+        self.verbose = verbose
+
+
+if __name__ == '__main__':
+    desc='Remove swap partitions from the root disk and reclaim their space. \
+          All arguments are for testing or debugging purposes only.'
+    parser = argparse.ArgumentParser(description=desc)
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument('-p', '--partition',
+        help='Partition on the disk where swap partitions will be removed.')
```

Comment:
it's a nitpick but argument help should start with lowercase and have no trailing `.` to match the built-in `--help` argument's help

Reply:
Agreed, I didn't notice the lack of consistency.